packages/python/plotly/plotly/tests/test_core/test_errors/test_dict_path_errors.py

```diff
@@ -0,0 +1,223 @@
+import plotly.graph_objects as go
+from _plotly_utils.exceptions import PlotlyKeyError
+import pytest
+
+
+def error_substr(s, r):
+    """ remove a part of the error message we don't want to compare """
+    return s.replace(r, "")
+
+
+@pytest.fixture
+def some_fig():
+    fig = go.Figure()
+    fig.add_trace(go.Scatter(x=[], y=[]))
+    fig.add_shape(type="rect", x0=1, x1=2, y0=3, y1=4)
+    fig.add_shape(type="rect", x0=10, x1=20, y0=30, y1=40)
+    fig.add_trace(go.Scatter(x=[1, 2, 3], y=[4, 5, 6]))
+    return fig
+
+
+def test_raises_on_bad_index(some_fig):
+    # Check indexing errors can be detected when path used as key to go.Figure
+    try:
+        x0 = some_fig["layout.shapes[2].x0"]
+    except IndexError as e:
```

Comment:
Ah, so every test here needs to be rewritten to use `with pytest.raises(...)` because otherwise these tests can pass if no error is raised! We need to guarantee that the error is raised here, and check its contents.

Reply:
I tried using pytest.raisesRegex because I wanted to check the error message, but had trouble making a regex matching multiple lines. I can show you the error when using the multiline flag for `re`, it kind of looked like a bug with `re`. Also I don't need a full-blown regex, just a string comparison would suffice.